notifier/notifier.go

```diff
@@ -490,13 +513,40 @@ func (n *Manager) sendAll(alerts ...*Alert) bool {
 				wg.Done()
 			}(ams, am)
 		}
-		ams.mtx.RUnlock()
```

Comment:
This would unlock the read lock, even though the spawned go routines still had references to `ams`, resulting in possible race conditions. Instead, the lock id dropped at the end of the function via `defer`. This is safe as the function waits for all spawned goroutines before exiting.

Reply:
Another option would be to still release the lock here and pass the necessary inputs to the goroutine:

```golang
ams.mtx.RLock()
for _, am := range ams.ams {
   ...
   go func(c *http.Client, u string) {
       if err := n.sendOne(ctx, c, u, b); err != nil {
       ...
      }
   }(ams.client, am.url().String())
}
ams.mtx.RUnlock()
```

Reply:
That's even better, thanks @simonpasquier.